### DISCUSSION TITLE
How to pass a list of regex in allow_origin_regex in FastAPI CORS Middleware?

### DISCUSSION THREAD

**Discussion by raghavsikaria** [fastapi/fastapi#8486] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
-
```

### Description

So, I have a list of very long and complicated regex which are all for my possible _origins_ for my FastAPI APP. I am using the CORS Middleware as given here: https://fastapi.tiangolo.com/tutorial/cors/#use-corsmiddleware (usage given below)

However,  **allow_origin_regex** argument takes only a _string_ as input
and *allow_origins* argument takes a list of strings as input which I would ideally want for **allow_origin_regex** as well.

So while I can do this:
```python
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

origins = [
    "http://localhost",
    "http://localhost:8080",
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

```

I can not do this:
```python
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

origins_regex = [
    "regex1",
    "regex2",
]

app.add_middleware(
    CORSMiddleware,
    allow_origin_regex=origins_regex,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

```

How can I achieve this functionality?

I inspected CORS code too in Starlette rep and it confirms about the arguments here: https://github.com/encode/starlette/blob/master/starlette/middleware/cors.py

Code snippet:
```python
class CORSMiddleware:
    def __init__(
        self,
        app: ASGIApp,
        allow_origins: typing.Sequence[str] = (),
        allow_methods: typing.Sequence[str] = ("GET",),
        allow_headers: typing.Sequence[str] = (),
        allow_credentials: bool = False,
        allow_origin_regex: typing.Optional[str] = None,
        expose_headers: typing.Sequence[str] = (),
        max_age: int = 600,
    ) -> None:
```

### Operating System

Linux

### FastAPI Version

0.75.0

### Python Version

Python 3.6.8

**Comment by aboubacs** [comment, marked as answer]
In your case, what's wrong with writing a regex that would match any of your origins ? (so in your example something like `"(regex1)|(regex2)"`)

**Comment by raghavsikaria** [comment]
There's nothing wrong, infact, that is what I am doing right now. It's just that, that feature will be much more cleaner. That option **exists in Django framework** where I have sent all regex as list. I find combining all long complicated regex in one string as unreadable.

Is this even a valid painpoint/usecase ? If not, I'll close/delete the issue.

**Comment by aboubacs** [comment]
I don't know, but to find out, it would probably be better to submit the issue in starlette. The `CORSMiddleware` from `fastapi.middleware.cors` is just the class from starlette but packaged in another import path (https://github.com/tiangolo/fastapi/blob/master/fastapi/middleware/cors.py), so the change to add the option to send a list of regex would only need to be done in starlette, not fastapi.
